src/main/java/io/cdap/plugin/gcp/bigquery/sink/AvroRecordWriter.java

```diff
@@ -101,7 +101,9 @@ private void createFileWriter(Schema writerSchema) throws IOException {
   /** {@inheritDoc} */
   @Override
   public void close(TaskAttemptContext context) throws IOException {
-    mAvroFileWriter.close();
+    if (mAvroFileWriter != null) {
```

Comment:
a similar fix should be done in the sync() method

Reply:
I added null check for sync() method. However sync() method has return value, in case when v is null this method will be return 0. Is it acceptable solution?

Reply:
yeah that's fine